app/src/androidTest/java/ch/epfl/sdp/drone3d/SaveMappingMissionActivityTest.kt

```diff
@@ -0,0 +1,203 @@
+        val name = "My mission"
+
+        val expectedMappingMission = MappingMission(name, flightPath)
+
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            SaveMappingMissionActivity::class.java
+        )
+        intent.putExtra("flightPath", flightPath)
+        ActivityScenario.launch<Activity>(intent)
+
+        onView(ViewMatchers.withId(R.id.missionName)).perform(ViewActions.click()).perform(
+            ViewActions.typeText(
+                name
+            )
+        ).perform(ViewActions.closeSoftKeyboard())
+
+        onView(ViewMatchers.withId(R.id.privateCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton)).perform(ViewActions.click())
+
+        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, expectedMappingMission)
+    }
+
+
+    @Test
+    fun checkBoxesEnableSaveButton() {
+        onView(ViewMatchers.withId(R.id.privateCheckBox))
+            .check(ViewAssertions.matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+        onView(ViewMatchers.withId(R.id.sharedCheckBox))
+            .check(ViewAssertions.matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(not(ViewMatchers.isEnabled())))
+
+
+        // Check private box
+        onView(ViewMatchers.withId(R.id.privateCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(ViewMatchers.isEnabled()))
+
+        // Uncheck private box
+        onView(ViewMatchers.withId(R.id.privateCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(not(ViewMatchers.isEnabled())))
+
+        // Check share box
+        onView(ViewMatchers.withId(R.id.sharedCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(ViewMatchers.isEnabled()))
+
+        // Uncheck share box
+        onView(ViewMatchers.withId(R.id.sharedCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(not(ViewMatchers.isEnabled())))
+
+        // Check both boxes
+        onView(ViewMatchers.withId(R.id.privateCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.sharedCheckBox)).perform(ViewActions.click())
+        onView(ViewMatchers.withId(R.id.saveButton))
+            .check(ViewAssertions.matches(ViewMatchers.isEnabled()))
+
+        // Uncheck share box
+        onView(ViewMatchers.withId(R.id.sharedCheckBox)).perform(ViewActions.click())
```

Comment:
You should import ViewMatchers.* and ViewAssertions.* and maybe ViewActions.* too